### DISCUSSION TITLE
Target-Typed New:  Infer Type Parameters

### DISCUSSION THREAD

**Discussion by TonyValenti** [dotnet/csharplang#4474] [Language Ideas] [answered]
It would be really great if I could do this:
```
//List<> becomes List<LicenseTemplateStorage>()
public virtual ICollection<LicenseTemplateStorage> LicenseTemplates { get; set; } = new List<>();
```

**Comment by huoyaoyuan** [comment]
It may be hard to decide for complex scenarios with composing and covariance, but only allowing the simplest form is also OK.

**Comment by svick** [comment, marked as answer]
This is a duplicate of https://github.com/dotnet/csharplang/discussions/2935.
